docs/reference/esql/functions/types/coalesce.asciidoc

```diff
@@ -6,8 +6,13 @@
 |===
 first | rest | result
 boolean | boolean | boolean
+boolean | | boolean
 integer | integer | integer
+integer | | integer
 keyword | keyword | keyword
+keyword | | keyword
 long | long | long
+long | | long
 text | text | text
+text | | text
```

Comment:
Hm, the type tables for functions that take 1 or more args of the same type actually look less useful now.

For instance, this is the type table for greatest:
![image](https://github.com/elastic/elasticsearch/assets/88324834/b93f3cc5-efed-4860-934b-3af152cf45df)

Maybe we should just special case such functions?

Reply:
It seems like it exposed that we miss some test cases for double, ip and version for variadic functions like greatest, perhaps not to add entries for variadic functions if the List argument has 0 or more than 1 items in it?

Reply:
We could, but that would imply that a "rest" is mandatory. Calling these functions with just one parameter is valid (and now documented, so IMO the table is now _more_ useful).
I think the issue is that the first and rest are of the same type, in this case, in which we know the function makes a pick based on value, irrespective of type (for those supported types). But looking over more (types of) functions, this case wouldn't be outstanding, IMO.

However, I'm OK with special-casing these functions if someone has a strong feeling about them.